src/plugins/kubeconfig/install.ts

```diff
@@ -0,0 +1,70 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import {
+  AwsInstall,
+  AwsItems,
+  ensureInstall,
+  InstallMetadata,
+} from "../../common/install";
+
+const EksItems = [...AwsItems, "kubectl"] as const;
+type EksItem = (typeof EksItems)[number];
+
+/**
+ * Converts the current system architecture, as represented in TypeScript, to
+ * the value used in the kubectl download URL, or throw an exception if the
+ * current architecture is not one kubectl has an official build for.
+ */
+const kubectlDownloadArch = (): string => {
+  const arch = process.arch;
+
+  switch (arch) {
+    case "x64": // macOS, Linux, and Windows
+      return "amd64";
+    case "arm64": // macOS and Linux only
+      return arch;
+    default:
+      throw `Unsupported system architecture for kubectl: ${arch}. Please install kubectl manually, or check that it is available in your PATH.`;
+  }
+};
+
+const kubectlInstallCommandsDarwin = (): Readonly<string[]> => {
```

Comment:
We not only give these commands for the AWS CLU and AWS Session Manager Plugin currently (see [here](https://github.com/p0-security/p0cli/blob/2ae3841b9ae920cbee4f0dd895964270af85ab31/src/plugins/aws/ssm/install.ts)), but we attempt to automatically perform the install if possible too. This just replicates that behavior, but for `kubectl`. Indeed, we can actually give better instructions than the official website can, because system architecture information is not conveyed in User-Agent information or in HTTP Client Hints. Indeed, visiting the [official instructions for installing `kubectl` on macOS](https://kubernetes.io/docs/tasks/tools/install-kubectl-macos/) shows the Intel instructions by default, even though I'm on an Apple Silicon machine.